src/utils/Api/fetchCars.ts

```diff
@@ -0,0 +1,58 @@
+import { CarsState } from "../../components/molecules/CarsContext";
+import z from "zod";
+import { CarSchema } from "../types/Car";
+
+/**
+ * Fetches cars from the server and sets the state accordingly.
+ * 
+ * @param state The state to set
+ * @param setState The state setter
+ */
+export default async function fetchCars(state: CarsState, setState: React.Dispatch<React.SetStateAction<CarsState>>) {
+    const response = await fetch("https://themikkel.dk/unfollow/sdu/cars/cars");
+
+    // Validate correct status code from server
+    if (!response.ok) {
+        setState({
+            cars: [],
+            error: `Error: ${response.status} - ${response.statusText}`,
+            loading: false,
+        });
+        return;
+    };
+
+    // Validate correct content type from server
+    const contentType = response.headers.get("Content-Type");
+    if (!contentType || !contentType.includes("application/json")) {
+        setState({
+            cars: [],
+            error: `Error: Invalid content type: ${contentType}`,
+            loading: false,
+        });
+        return;
+    }
+
+    // Parse response body as json
+    const cars = await response.json() as unknown;
```

Comment:
Seems hella sus with unknown for type safety concerns, but seems like there are no fixes...
https://github.com/node-fetch/node-fetch/issues/1262

Reply:
This have been discussed, and is the correct implementation.